### PR TITLE
Sheer Force fix and move effect cleanup

### LINKED ISSUE #5800: Eevee LGPE move issues
### Description

When Glitzy Glow fails, it still sets up the status.

Eevee Starter moves that have a dedicated effect occasionally print the wrong string.

### Version

1.10.0 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

_No response_

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes #5800

Plasma Fist and Eevee signature moves set up their effects in some cases even if they failed and the message was occasionally wrong. This is because their scripts were written as if they were status moves. 

This pr changes the effects to move effects which fixes the problems and makes them functionally the same as something like `MOVE_EFFECT_BURN` 

The move Sparkly Swirl wasn't fixed because Aromatherapy doesn't seem to have the correct effect in the first place and would be out of scope for this pr imo. It is probably best to open a new issue for this once this pr is merged. 

## People who collaborated with me
Tests provided by @hedara90

### CODE REVIEW

File: src/data/moves_info.h [hunk 1]
```diff
@@ -16772,6 +16772,10 @@ const struct MoveInfo gMovesInfo[MOVES_COUNT_DYNAMAX] =
         .contestCategory = CONTEST_CATEGORY_COOL,
         .contestComboStarterId = 0,
         .contestComboMoves = {0},
+        .additionalEffects = ADDITIONAL_EFFECTS({
+            .moveEffect = MOVE_EFFECT_ION_DELUGE,
+            .chance = 100,
+        }),
```
Comment (hedara90) [inline src/data/moves_info.h:16778]: This makes Plasma Fists get boosted by Sheer Force, which as far as I know isn't how it should work.
Tested on Showdown due to lack of access to Zeraora on cartridge, but I can't find Plasma Fists in the list of moves boosted by Sheer Force.

**Comment by AlexOn1ine** [comment]
Further fixes:

- Plasma Fists does not receive the Sheer Force boost anymore
- Salt Cure and Eerie Spell are now Sheer Force boosted
- Malignant Chain now receives the Sheer Force boost but needs a proper implementation. See comment.

Tests provided by @hedara90

File: test/battle/move_effect_secondary/reflect.c [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            ANIMATION(ANIM_TYPE_MOVE, MOVE_BADDY_BAD, player);
+            MESSAGE("Reflect made your team stronger against physical moves!");
+        } else {
+            NONE_OF {
+                ANIMATION(ANIM_TYPE_MOVE, MOVE_BADDY_BAD, player);
+                MESSAGE("Reflect made your team stronger against physical moves!");
+            }
+        }
+
```
Comment (hedara90) [inline test/battle/move_effect_secondary/reflect.c:31]:
```suggestion
```

File: test/battle/move_effect_secondary/light_screen.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            ANIMATION(ANIM_TYPE_MOVE, MOVE_GLITZY_GLOW, player);
+            MESSAGE("Light Screen made your team stronger against special moves!");
+        } else {
+            NONE_OF {
+                ANIMATION(ANIM_TYPE_MOVE, MOVE_GLITZY_GLOW, player);
+                MESSAGE("Light Screen made your team stronger against special moves!");
+            }
+        }
+
```
Comment (hedara90) [inline test/battle/move_effect_secondary/light_screen.c:31]:
```suggestion
```

File: test/battle/move_effect_secondary/haze.c [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        {
+            ANIMATION(ANIM_TYPE_MOVE, MOVE_FREEZY_FROST, player);
+            MESSAGE("All stat changes were eliminated!");
+        } else {
+            NONE_OF {
+                ANIMATION(ANIM_TYPE_MOVE, MOVE_FREEZY_FROST, player);
+                MESSAGE("All stat changes were eliminated!");
+            }
+        }
+
```
Comment (hedara90) [inline test/battle/move_effect_secondary/haze.c:31]:
```suggestion
```

File: test/battle/ability/sheer_force.c [hunk 5]
```diff
@@ -7,64 +7,908 @@ ASSUMPTIONS
...
+        case MOVE_WICKED_TORQUE:
+        case MOVE_WILDBOLT_STORM:
+        case MOVE_ZING_ZAP:
+        case MOVE_ELECTRO_SHOT:
+        case MOVE_PSYCHIC_NOISE:
+            return TRUE;
+    }
+    return FALSE;
+}
+
```
Comment (hedara90) [inline test/battle/ability/sheer_force.c:612]:
```suggestion
//  Test split into four parts that handles ~1/4 of all moves each
```

**Review by hedara90 — COMMENTED** [review]
Minor formatting, will look at it properly tomorrow

**Review by hedara90 — APPROVED** [review]

**Pull request merged**